spec/support/vmdb_configuration_helper.rb

```diff
@@ -3,6 +3,7 @@ def stub_server_configuration(config, config_name = "vmdb")
     configuration = double(:config                         => config,
                            :merge_from_template_if_missing => nil,
                            :merge_from_template            => nil)
+    allow(configuration).to receive(:fetch_with_fallback).and_return { |*arg| config.fetch_path(arg) }
```

Comment:
If we are stubbing out the configuration and just using a hash, `merge_from_template` and `merge_from_template_if_missing` are no-ops and handled on line 4,5.

The specs all pass. Implementing configuration override would just introduce complexity. (That level of detail is best tested in `configuration_spec.rb`.)